File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.ArrayList;

class Solution {
    
    char[] vowel = {'A', 'E', 'I', 'O', 'U'};
    ArrayList<String> words = new ArrayList<>();
    
    public int solution(String word) {
        
        dfs(0, "");
        
        return words.indexOf(word);
    }
    
    public void dfs(int depth, String s){
        
        words.add(s);
        
        if(depth == 5) {
            return;
        }
        
        for(char c : vowel){
            dfs(depth + 1, s + c);
        }
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
vowels = ['A', 'E', 'I', 'O', 'U']

def solve(word, words):
    
    words.append(word)
    
    if len(word) == 5:
        return
    
    for c in vowels:
        solve(word + c, words)

def solution(word):
    words = []
    
    solve("", words)
    
    return words.index(word)